index_documents.go

```diff
@@ -23,14 +49,14 @@ func sendCsvRecords(documentsCsvFunc func(recs []byte, pk ...string) (resp *Task
 		return nil, fmt.Errorf("could not write CSV records: %w", err)
 	}
 
-	resp, err := documentsCsvFunc(b.Bytes(), primaryKey...)
+	resp, err := documentsCsvFunc(b.Bytes(), options)
 	if err != nil {
 		return nil, err
 	}
 	return resp, nil
 }
 
-func (i Index) saveDocumentsFromReaderInBatches(documents io.Reader, batchSize int, documentsCsvFunc func(recs []byte, pk ...string) (resp *TaskInfo, err error), primaryKey ...string) (resp []TaskInfo, err error) {
+func (i Index) saveDocumentsFromReaderInBatches(documents io.Reader, batchSize int, documentsCsvFunc func(recs []byte, op *CsvDocumentsQuery) (resp *TaskInfo, err error), options *CsvDocumentsQuery) (resp []TaskInfo, err error) {
```

Comment:
Is this method only used for CSV?

Reply:
Yes

Reply:
Ok! It was not very clear per the function name